source/main.cpp

```diff
@@ -13,44 +16,53 @@
 #include "parser/parser.h"
 
 int main(int argc, char ** argv) {
+  // Install the segfault handler with backtrace
+  signal(SIGSEGV, handler);
 
-// The valid command line usage is "./mc-mini <parameter file>". Otherwise, throw an exception.
-  if (argc == 1) {
-    throw std::invalid_argument("usage: " + std::string{argv[0]} + " <parameter file>.");
-  }
+  try {
+    // The valid command line usage is "./mc-mini <parameter file>". Otherwise, throw an exception.
+    if (argc == 1) {
+      THROW_WITH_TRACE(InvalidArgument() <<
+              errmsg_info("usage: " + static_cast<std::string>(argv[0]) + " <parameter file>."));
```

Comment:
With `std::string` it doesn't appear to make much of a difference. Using `std::string{argv[0]}` returns the same result as using `static_cast<std::string>(argv[0])` or `std::string(argv[0])` or `(std::string) argv[0]`. There are some benefits to using `static_cast` over other methods (see [here](http://stackoverflow.com/questions/28002/regular-cast-vs-static-cast-vs-dynamic-cast) for more info), but the main reason I used it here is because `static_cast` is somewhat more noticeable/readable, and also because my compiler was giving me warnings about the brace-initializer list.